reframe/listener.go

```diff
@@ -400,8 +405,15 @@ func (listener *ReframeListener) notifyRemoveAndPersist(ctx context.Context, chu
 	log.Infof("Notifying Remove for chunk=%s", ctxIdStr)
 
 	// notifying the indexer
-	_, err := listener.engine.NotifyRemove(ctx, listener.provider(), chunk.ContextID)
-	if err != nil && err != provider.ErrAlreadyAdvertised {
+	err := RetryWithBackoff(func() error {
+		_, e := listener.engine.NotifyRemove(ctx, listener.provider(), chunk.ContextID)
+		if e == nil || e == provider.ErrAlreadyAdvertised {
+			return nil
+		}
+		return e
```

Comment:
style nit: Slightly more clear that a specific error is being suppressed.
```suggestion
		if e == provider.ErrAlreadyAdvertised {
			e = nil
		}
		return e
```

Reply:
changed